drivers/bluetooth/hci/hci_nxp.c

```diff
@@ -346,13 +355,16 @@ static int bt_nxp_open(void)
 			break;
 		}
 #endif
+
+		hci->recv = recv;
```

Comment:
@jhedberg Could you please move this to be called before` bt_nxp_set_calibration_data()`? This function is sending HCI messages so hci_rx_cb() gets called and uses this `recv` which still points to `NULL`.

Beside this, I did a sanity check on `rd_rw612_bga` and it worked fine with this fix. 

Thanks for the effort.

Reply:
@yeaissa thanks for testing this. The reason I didn't catch this particular issue, is that it's not really valid for the driver's `open()` callback to try to send HCI data, at least not using `bt_hci_cmd_send*()` APIs. From the host's perspective open() is what's used to "open" the HCI transport, and it will assume that the transport is not available until that driver callback has returned success. It's by chance that we don't have any checks or host-side state machine initializations tied to this, and so the existing code just works. The right place to do HCI command sending is instead the `setup()` callback - that's really the purpose of it, i.e. anything the HCI driver might need to do before the host sends its first HCI packet.

Reply:
@jhedberg, I understand your point and it makes sense. I'll update in a separate PR. Thanks.
I guess for now you can just move it in the meantime.

Reply:
@yeaissa thanks. I incorporated your commit from #73711 into this one (it's the first commit here). I'll rebase once #73711 has been merged.

Reply:
Thanks @jhedberg, It's merged now.